android/source/VideoLocker/src/org/edx/mobile/view/MyCoursesListActivity.java

```diff
@@ -323,4 +337,51 @@ public void onFail(Exception ex) {
             logger.error(ex);
         }
     };
+
+    /**
+     * Checks progress of all the videos that are being downloaded.
+     * If progress of any of the downloads is 100%, then marks the video as DOWNLOADED.
+     */
+    private void doRepairData() {
+        Thread maintenanceThread = new Thread(new Runnable() {
```

Comment:
Yes, <code>getApplicationContext()</code> works for database operations and will work even if activity goes off.